clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -1595,4 +1595,105 @@ else if (recordPartition != null)
             return topicPartition;
         }
     }
+
+    /**
+     * A class that represents the state of a prepared transaction that can be
+     * serialized to and deserialized from a string.
+     * Objects of this class can be written to and read from a database
+     * to support recovery of prepared transactions.
+     */
+    public static class PreparedTxnState {
+        private final long producerId;
+        private final short epoch;
+        
+        /**
+         * Creates a new empty PreparedTxnState
+         */
+        public PreparedTxnState() {
+            this.producerId = -1L;
+            this.epoch = -1;
+        }
+        
+        /**
+         * Creates a new PreparedTxnState from a serialized string representation
+         * 
+         * @param serializedState               The serialized string to deserialize.
+         * @throws IllegalArgumentException if the serialized string is not in the expected format
+         */
+        public PreparedTxnState(String serializedState) {
+            if (serializedState == null || serializedState.isEmpty()) {
+                this.producerId = -1L;
+                this.epoch = -1;
+                return;
+            }
+            
+            try {
+                String[] parts = serializedState.split(":");
+                if (parts.length != 2) {
+                    throw new IllegalArgumentException("Invalid serialized transaction state format: " + serializedState);
+                }
+                
+                this.producerId = Long.parseLong(parts[0]);
+                this.epoch = Short.parseShort(parts[1]);
+            } catch (NumberFormatException e) {
+                throw new IllegalArgumentException("Invalid serialized transaction state format: " + serializedState, e);
+            }
+        }
+        
+        /**
+         * Creates a new PreparedTxnState with the given producer ID and epoch
+         * 
+         * @param producerId        The producer ID
+         * @param epoch             The producer epoch
+         */
+        public PreparedTxnState(long producerId, short epoch) {
```

Comment:
This shouldn't be public.  An application should be able to either create `PreparedTxnState()` or `PreparedTxnState(string)`.